### PR TITLE
Integrating Stripe into Formspree

### PR DESCRIPTION
**Pull request by smakosh**
See: https://github.com/formspree/formspree-react/pull/10

### CODE REVIEW

File: src/index.ts
Comment (colevscode) [inline src/index.ts:91]: Minor nit: I'm a little confused why we call this handlePayment. Doesn't it just create a paymentMethod so should be called createPaymentMethod or something?

Comment (smakosh) [reply src/index.ts:91]: Sure I can rename it

**Comment by colevscode** [comment]
I think we should change SuccessBody to:

```
export interface SuccessBody {
  next?: string;
}
```

The `ok` property is redundant to the status code, and should be deprecated. We can do so by just leaving it out of the type.

The original `id` and `data` properties were included for compatibility with existing statickit implementations back when we merged, but we never send either of those properties, so we should just remove them.

I don't think we send any other data for a successful POST to the endpoint.

File: src/index.ts
Comment (smakosh) [reply src/index.ts:194]: Will be handled once the backend returns the right response

File: src/index.ts
Comment (smakosh) [reply src/index.ts]: Will also handle this once the server response is handled

File: src/index.ts
Comment (smakosh) [reply src/index.ts]: Agree.

File: src/forms.ts
Comment (rohitdatta) [inline src/forms.ts:39]: I think we can remove this todo now

File: src/forms.ts
Comment (rohitdatta) [inline src/forms.ts:43]: stripeCode should also be optional I think. There could be a plugin in the future that uses a different key

File: src/forms.ts
Comment (colevscode) [inline src/forms.ts:23]: What if we used a "STRIPE_CLIENT_ERROR" instead and used it for both payment method errors and stripe promise errors?

File: src/forms.ts
Comment (colevscode) [inline src/forms.ts:40]: I don't think we need the `| string` anymore if we're now using the details block.

File: src/forms.ts
Comment (colevscode) [inline src/forms.ts:89]: Can this be 

`response?: Response`

?

File: src/index.ts
Comment (colevscode) [inline src/index.ts:21]: Not sure why sometimes we use `?` and sometimes `| undefined`. Feel like we should just standardize around `project?: string` rather than `project: string | undefined`

File: src/index.ts
Comment (rohitdatta) [inline src/index.ts:99]: does the server not provide this payload itself? If so, can we just return the response from the server which should be correctly formatted like so

File: src/index.ts
Comment (colevscode) [inline src/index.ts:136]: Why do you need to check and catch these cases? In both the default success and error cases, you can just return

```
return { response, body: responseData };
```

File: src/index.ts
Comment (colevscode) [inline src/index.ts:138]: Can we break out this handleSCA stuff into a separate function to make the flow clearer?

File: src/index.ts
Comment (colevscode) [inline src/index.ts:196]: Rather than calling `appendExtraData`, then `fetchImpl` and finally handling errors / success (again) could we just call `submitForm()` again here, passing in the extra data to the `data` argument, and removing the `createPaymentMethod` opt?

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:23]: True, will work on it!

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:40]: Yup correct

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:89]: I'll check

File: src/index.ts
Comment (smakosh) [reply src/index.ts:21]: `?` means it's optional, and using `| undefined` means it's required but can undefined.

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:43]: True

File: src/index.ts
Comment (smakosh) [reply src/index.ts:196]: I try to avoid recursive functions as much as I can 😆

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:89]: If Stripe fails to create a payment method, that's when we return a null response.

File: src/forms.ts
Comment (smakosh) [reply src/forms.ts:89]: https://github.com/formspree/formspree-core/pull/7#discussion_r906732550

File: src/index.ts
Comment (smakosh) [reply src/index.ts:99]: This is not a server side response, this is from Stripe itself.

File: src/forms.ts
Comment (colevscode) [inline src/forms.ts:21]: We should keep `STRIPE_SCA_ERROR` in this list since it can be sent from the server.

**Review by colevscode — APPROVED** [review]
It's a lot cleaner now. Looking good. Just add back in the enum and I think it can 🚢

**Review by rohitdatta — COMMENTED** [review]
Looks like the CI is failing with the same conflict I was getting locally. LGTM once tests pass

**Review by rohitdatta — APPROVED** [review]

**Comment by rohitdatta** [comment]
Good with me. I think this is worthy of a minor version upgrade, which we can release after you merge

**Pull request merged**
